packages/legend-query-builder/src/components/__tests__/QueryBuilderFunctionExplorerTree.test.tsx

```diff
@@ -0,0 +1,187 @@
+    );
+    const graphManagerState = queryBuilderState.graphManagerState;
+    graphManagerState.graph.dependencyManager = dependencyManager;
+    await graphManagerState.graphManager.buildDependencies(
+      graphManagerState.coreModel,
+      graphManagerState.systemModel,
+      dependencyManager,
+      dependencyEntitiesIndex,
+      graphManagerState.dependenciesBuildState,
+    );
+    expect(graphManagerState.dependenciesBuildState.hasSucceeded).toBe(true);
+
+    const employeeClass =
+      queryBuilderState.graphManagerState.graph.getClass('test::Employee');
+    await act(async () => {
+      queryBuilderState.changeClass(employeeClass);
+    });
+    await act(async () => {
+      queryBuilderState.setShowFunctionsExplorerPanel(true);
+    });
+    const queryBuilderFunctionPanel = await waitFor(() =>
+      renderResult.getByTestId(QUERY_BUILDER_TEST_ID.QUERY_BUILDER_FUNCTIONS),
+    );
+    expect(
+      getByText(queryBuilderFunctionPanel, 'test::testYtd():Any[*]'),
+    ).not.toBeNull();
+    fireEvent.click(
+      getByTitle(queryBuilderFunctionPanel, 'Show Options Menu...'),
+    );
+    fireEvent.click(
+      renderResult.getByText('Show functions from dependency projects'),
+    );
+    expect(
+      getByText(queryBuilderFunctionPanel, 'domain::testFunction():String[1]'),
+    ).not.toBeNull();
+    expect(
+      getByText(
+        queryBuilderFunctionPanel,
+        'domain::my::testFunction2():String[1]',
+      ),
+    ).not.toBeNull();
+    fireEvent.click(getByTitle(queryBuilderFunctionPanel, 'View as Tree'));
+    expect(getByText(queryBuilderFunctionPanel, 'test')).not.toBeNull();
+    fireEvent.click(getByText(queryBuilderFunctionPanel, 'test'));
+    expect(
+      getByText(queryBuilderFunctionPanel, 'testYtd():Any[*]'),
+    ).not.toBeNull();
+    expect(getByText(queryBuilderFunctionPanel, 'domain')).not.toBeNull();
+    fireEvent.click(getByTitle(queryBuilderFunctionPanel, 'domain'));
+    expect(
+      getByText(queryBuilderFunctionPanel, 'testFunction():String[1]'),
+    ).not.toBeNull();
+    fireEvent.click(getByTitle(queryBuilderFunctionPanel, 'my'));
+    expect(
+      getByText(queryBuilderFunctionPanel, 'testFunction2():String[1]'),
+    ).not.toBeNull();
+
+    //drag and drop
+    fireEvent.click(getByTitle(queryBuilderFunctionPanel, 'View as List'));
+    const tdsProjectionPanel = await waitFor(() =>
+      renderResult.getByTestId(
+        QUERY_BUILDER_TEST_ID.QUERY_BUILDER_TDS_PROJECTION,
+      ),
+    );
+    const tdsProjectionDropZone = await waitFor(() =>
+      getByText(tdsProjectionPanel, 'Add a projection column'),
+    );
+    const ytdFunction = await waitFor(() =>
+      getByText(queryBuilderFunctionPanel, 'test::testYtd():Any[*]'),
+    );
+    await dragAndDrop(
+      ytdFunction,
+      tdsProjectionDropZone,
+      tdsProjectionPanel,
+      'Add a projection column',
+    );
+    await waitFor(() => getByText(tdsProjectionPanel, 'x|test::testYtd()'));
```

Comment:
Nice! out of curiosity, why does this work? I thought we would need to mock monaco editor here ?